src/utils/api/callout.ts

```diff
@@ -53,6 +54,17 @@ export async function fetchCallout(id: string): Promise<GetMoreCalloutData> {
   return toCallout(data);
 }
 
+export async function createCallout(
+  calloutData: CreateCalloutData
+): Promise<GetBasicCalloutData> {
+  const { data } = await axios.post<Serial<GetBasicCalloutData>>(
+    '/callout',
+    // TODO: passing calloutData directly is not safe, it could contain extra properties
+    calloutData
+  );
+  return toCallout(data);
+}
+
```

Comment:
That will do for now!

Reply:
Would love to talk about safety as I'm not 100% clear about the risks, given that the requests appears to require admin auth and [casts the payload into our poll data structure.](https://github.com/beabee-communityrm/beabee/blob/c64684c6c65c33595904cbf9460ca99da577f457/src/api/controllers/CalloutController.ts#L34-L51)

Reply:
It's not a security risk, but the API will reject any request that contains extra properties, e.g.

```ts
// Accepted
{
  "name": "blah"
}

// Rejected
{
  "name": "blah",
  "foo": "bar"
}
```

Unfortunately there isn't a way in TypeScript to say the object must be **just** the properties in `CreateCalloutData`, so `calloutData` could have extra properties and passing it directly to the API means it would then be rejected. We should really rebuild the payload for the request, but that is a lot of boilerplate! Maybe there is a better way